lib/Yam/Installation/Upgrade/PreviousRepoPage.pm

```diff
@@ -0,0 +1,27 @@
+# SUSE's openQA tests
+#
+# Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Handle page that allows to select system/partition to be upgraded.
+# Maintainer: QE YaST and Migration (QE Yam) <qe-yam at suse de>
+
+package Yam::Installation::Upgrade::PreviousRepoPage;
+use parent 'Installation::Navigation::NavigationBase';
```

Comment:
we are going to avoid inheritance at all costs for new designs regarding UI elements, but we can inherit here from PageBase and avoid to pass `app => ...` in `lib/Distribution/Opensuse/Tumbleweed.pm` for structural reasons, For compose UI elements we will use composition, but for now we can just remap `press_next`.
It might be some cases where the button needs to be enable or we have to wait for something, so we will specifically wait for that thing in the test level or we would put in its own page the bottom part with the help the next etc. The previous design was not really transparent IMO, the POM is exclusively a repository for locators, the composition or inheritance of POM is another problem to solve always in each design, so as far as we are not really sure, we can just make those POM very trivial following the pattern by the book.

Reply:
I assume "remap press_next" means to put it in each page ? So we'll repeat the same in each page ?

Reply:
I think this will likely cause the issues that made us iplement this in the past, and we'll have to apply the same fix in every single module, no ?

Reply:
yes, it is simple enough to repeat this code for next to be acceptable for now (now we are rewriting to not use controllers and it is an opportunity to start to make it better).

A better option is to have a page called Navigation for mapping Help, Abort, Back, Next.
In Java or in Perl composition can only be done by re-mapping, so you still need a call to a method ->next() which use a method from object Navigation. The problem is that this doesn't work if we detach the object to click next for example from the object that check if the page is there, when we click and the page is not there we would fail and the ribbon might be used with different pages. Then it would require something more elaborated.

We need some common interface method for all pages, so instead of `get_select_origin` all the methods like this should be called the same like `page_loaded()`. Then when you initialize the Navigation in one page, you need to construct this object passing it and instance of the class where is included `$self`,  here there is a draft for that:

```
package XX::YY::SelectForUpdatePage
$self->{Navigation} = XXX::YYY::Navigation->new($self);

sub press_next {
    my ($self) = @_;
    $self->{Navigation}->next();
}

package XXX::YYY::Navigation
sub new {
    my ($class, $args) = @_;
    $self->{base} = $args->{base};

sub next {
    my ($self) = @_;
    $self->{base}->page_loaded();
    $self->{btn_ok}->click();
}
```

We can always leave this point open for future implementation and go for the easy thing to do which is repeat the code for the next button, but we should avoid inheritance (for adding functionality, for structural reason is fine).